src/bus/mod.rs

```diff
@@ -164,8 +164,16 @@ impl Bus {
 
                 addr
             }
-            UnixSocket::Dir(_) => bail!("`dir` transport is not supported (yet)."),
-            UnixSocket::TmpDir(_) => bail!("`tmpdir` transport is not supported (yet)."),
+            UnixSocket::Dir(dir) | UnixSocket::TmpDir(dir) => {
+                let path = dir.join(format!("dbus-{}", fastrand::u32(1_000_000..u32::MAX)));
```

Comment:
I picked a minimum of 1M just so that the sequence of random characters would be at least 7 characters long :shrug: